execution/executor/src/components/block_tree/test.rs

```diff
@@ -0,0 +1,135 @@
+                None,
+            )
+            .unwrap();
+
+        Self {
+            root: Mutex::new(root),
+            block_lookup,
+        }
+    }
+
+    pub fn size(&self) -> usize {
+        self.block_lookup.inner.lock().0.len()
+    }
+}
+
+fn id(index: u64) -> HashValue {
+    let bytes = index.to_be_bytes();
+    let mut buf = [0; HashValue::LENGTH];
+    buf[HashValue::LENGTH - 8..].copy_from_slice(&bytes);
+    HashValue::new(buf)
+}
+
+fn empty_chunk() -> ExecutedChunk {
+    ExecutedChunk::new_empty(ExecutedTrees::new_empty())
+}
+
+fn gen_ledger_info(block_id: HashValue, reconfig: bool) -> LedgerInfo {
+    LedgerInfo::new(
+        BlockInfo::new(
+            1,
+            0,
+            block_id,
+            HashValue::zero(),
+            0,
+            0,
+            if reconfig {
+                Some(EpochState::empty())
+            } else {
+                None
+            },
+        ),
+        HashValue::zero(),
+    )
+}
+
+fn create_tree() -> BlockTree {
+    //    * ---> 1 ---> 2
+    //    |      |
+    //    |      └----> 3 ---> 4
+    //    |             |
+    //    |             └----> 5
+    //    |
+    //    └----> 6 ---> 7 ---> 8
+    //           |
+    //           └----> 9 ---> 10
+    //                  |
+    //                  └----> 11
+    // *: PRE_GENESIS_BLOCK_ID
+    let block_tree = BlockTree::new_empty();
+
+    block_tree
+        .add_block(*PRE_GENESIS_BLOCK_ID, id(1), empty_chunk())
+        .unwrap();
+    block_tree.add_block(id(1), id(2), empty_chunk()).unwrap();
+    block_tree.add_block(id(1), id(3), empty_chunk()).unwrap();
+    block_tree.add_block(id(3), id(4), empty_chunk()).unwrap();
+    block_tree.add_block(id(3), id(5), empty_chunk()).unwrap();
+    block_tree
+        .add_block(*PRE_GENESIS_BLOCK_ID, id(6), empty_chunk())
+        .unwrap();
+    block_tree.add_block(id(6), id(7), empty_chunk()).unwrap();
+    block_tree.add_block(id(7), id(8), empty_chunk()).unwrap();
+    block_tree.add_block(id(6), id(9), empty_chunk()).unwrap();
+    block_tree.add_block(id(9), id(10), empty_chunk()).unwrap();
+    block_tree.add_block(id(9), id(11), empty_chunk()).unwrap();
+    block_tree
+}
+
+#[test]
+fn test_branch() {
+    let block_tree = create_tree();
+    // put counting blocks as a separate line to avoid core dump
+    // if assertion fails.
+    let num_blocks = block_tree.size();
+    assert_eq!(num_blocks, 12);
+    block_tree.prune(&gen_ledger_info(id(9), false)).unwrap();
+    let num_blocks = block_tree.size();
+    assert_eq!(num_blocks, 3);
+    assert_eq!(block_tree.root_block().id, id(9));
+}
+
+#[test]
+fn test_reconfig_id_update() {
+    let block_tree = create_tree();
+    block_tree.prune(&gen_ledger_info(id(1), true)).unwrap();
+    let num_blocks = block_tree.size();
+    // reconfig suffix blocks are ditched
+    assert_eq!(num_blocks, 1);
+    assert_ne!(block_tree.root_block().id, id(1));
```

Comment:
the id should be the new genesis?

Reply:
yeah so it's assert_NE? tripped me as well. Let me see if I can do it right.